cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -118,16 +141,31 @@ struct Day {
 
 // ----------------------------------------------------------------------
 // Extract day of week from timestamp
+//
+// By default week starts on Monday represented by 0 and ends on Sunday represented
+// by 6. Start day of the week (Monday=1, Sunday=7) and numbering start (0 or 1) can be
+// set using DayOfWeekOptions
 
 template <typename Duration>
 struct DayOfWeek {
+  explicit DayOfWeek(DayOfWeekOptions options) : options(options) {
```

Comment:
Nit: can you usually pass options as `const&`.

Reply:
Done.